### ISSUE TITLE
php-gds vs. new official Datastore API?

### ISSUE THREAD

**Issue by ajck** [tomwalder/php-gds#146]
@tomwalder I've been an advocate of php-gds for some time both for my own co's projects and in occasional posts as a Google Cloud Insider (more direct access to some of the Google folks building their cloud), and really appreciate your work on this library.

Now that Google have started getting proper PHP support out of the door, wondering if I should/"am supposed to be" moving to their interface. https://googlecloudplatform.github.io/google-cloud-php/#/docs/v0.10.1/datastore/datastoreclient

Just wondering what your take was on things, please Tom? i.e. do you see php-gds as winding down and people moving to the official offering (esp whenever their protocol buffer support appears), or will you be wrapping their functionality into this lib? Are there any significant performance or feature advantages one way or the other?

Many thanks,

**Comment by tomwalder** [comment]
Hi there.

**TL;DR Hopefully some healthy competition, and more options for developers!**

I plan to maintain PHP-GDS well into the future. My company uses it in commercial applications - as do many others now.

I am very aware of the "official" Google client - in fact there was some initial involvement from me on that project. However, I didn't have the time to dedicate to re-write PHP-GDS for them, and it would have meant all current users would be "out in the dark".

So, I guess the long story short is..

There are 2 libraries now. PHP-GDS is more well established, with a larger existing user base. Those people will be supported, and I will do my best to move the library forward.

It will up be to individual developers and organisations to decide which library suits them best.

Cheers! And thanks for all your support so far!

**Comment by tomwalder** [comment]
Worth noting that, at the time of writing, PHP-GDS is by far the most performant option for AppEngine.

And they both use the same REST API when used off-AppEngine.

I think the Google library is probably a closer integration to the exposed APIs from Datastore - which may be a developer preference.

**Comment by ajck** [comment]
Great, thanks Tom, encouraging to hear all that. Why is PHP-GDS by far the most performant? Is it just because of the protocol buffer usage, and if so will they not be on par shortly when the official lib gets protocol buffer support, or are there other reasons?

Thanks again.

**Comment by tomwalder** [comment]
Last time I checked, the Google library only supports the REST API.

I believe they are planning to add support for the gRPC API. But That gRPC API won't currently run on AppEngine standard.
